power-sequencer/mihawk-cpld.cpp

```diff
@@ -217,6 +221,81 @@ void MIHAWKCPLD::onFailure()
 
 void MIHAWKCPLD::analyze()
 {
+    //analyze psu status when power_on fault.
+    auto poweronError = checkPoweronFault();
+    if(poweronError)
+    {
+        int errorcode;
+        errorcode = checkPSUDCpgood(true);
+        if (!((errorcode >> 1) & 1) && !((errorcode >> 3) & 1) )
+        {
+            report<PsuErrorCode0>();
+        }
+        else if (!((errorcode >> 2) & 1) && !((errorcode >> 4) & 1))
+        {
+            report<PsuErrorCode1>();
+        }
+    }
+}
+
+// Check for PoweronFault
+bool MIHAWKCPLD::checkPoweronFault()
+{
+    bool result = 0;
+    std::string i2cBus = "/dev/i2c-" + std::to_string(busId);
+
+    // open i2c device(CPLD-PSU-register table)
+    int fd = open(i2cBus.c_str(), O_RDWR | O_CLOEXEC);
+    if (fd < 0)
+    {
+        std::cerr << "Unable to open i2c device \n";
+    }
+
+    // set i2c slave address
+    if (ioctl(fd, I2C_SLAVE_FORCE, slaveAddr) < 0)
+    {
+        std::cerr << "Unable to set device address \n";
+        close(fd);
```

Comment:
if the open() failed above, none of these closes will do anything, and every one of these ioclts will fail.  I don't think it will actually affect the functionality though.

Reply:
@spinler I'll fix it in upstream.